### PR TITLE
Merge changes to top level CMakeLists.txt

### PR DESCRIPTION
**Pull request by JosiahWI**
Resolves: N/A
Authors:
@josiah_wi

## Summary of changes
Automatically update git submodules during build.
Change minimum CMake version to 3.12.
  
## On approval
Merge

### CODE REVIEW

**Comment by beeperdeeper089** [comment]
I'll take a look at this, and test it with CMake 3.18.1 (I use 3.16 rn but updating is always handy).

If we're cloning in the CMake, we should probably remove the part from the CI script.

**Comment by beeperdeeper089** [comment]
@JosiahWI CMake 3.18.1 works just fine, so I would remove the version range within the CMake. Just set the minimum and that should be fine. The Git thing should be fine, but i would say change the messages to:
```
Updating Git Submodules
and
Git Submodules will not be updated, manual checking out is required.
```
or something more descriptive than ``Not updating``

**Comment by JosiahWI** [comment]
The version range sets the correct policies if the CMake version is 3.12 or newer.

**Comment by beeperdeeper089** [comment]
```
cmake_minimum_required(VERSION 3.8...3.16)
```
Right, but this line can become:
```
cmake_minimum_required(VERSION 3.8)
```

File: azure-pipelines.yml [hunk 1]
```diff
@@ -44,9 +44,7 @@ jobs:
 #        key: 'ccache | "$(Agent.JobName)"'
 #        path: $(CCACHE_DIR)
 #      displayName: 'CCache'
-    - script: git submodule update --init
-      displayName: 'Cloning Submodules'
-    - script: cmake -H. -BBuild
+    - script: cmake -S . -B Build -D UPDATE_SUBMODULES=ON
```
Comment (beeperdeeper089) [inline azure-pipelines.yml:47]: If the option is on by default, do we need to define this?

Also, it's recommended to not use spaces in the CMake command, so like:
```
cmake -S. -BBuild -DUPDATE_SUBMODULES=ON
```

Comment (beeperdeeper089) [reply azure-pipelines.yml:47]: Ignore, me, the spaces is something that was an issue a while ago with things like Command Prompt on Windows which wouldn't work with spaces for some reason.

**Review by beeperdeeper089 — APPROVED** [review]

**Pull request merged**